tests/components/plex/test_config_flow.py

```diff
@@ -502,3 +530,194 @@ async def test_multiple_servers_with_import(hass):
         )
         assert result["type"] == "abort"
         assert result["reason"] == "non-interactive"
+
+
+async def test_manual_config(hass):
+    """Test creating via manual configuration."""
+
+    class WrongCertValidaitionException(requests.exceptions.SSLError):
+        """Mock the exception showing an unmatched error."""
+
+        def __init__(self):
+            self.__context__ = ssl.SSLCertVerificationError(
+                "some random message that doesn't match"
+            )
+
+    # Basic mode
+    result = await hass.config_entries.flow.async_init(
+        config_flow.DOMAIN, context={"source": "user"}
+    )
+
+    assert result["data_schema"] is None
+    hass.config_entries.flow.async_abort(result["flow_id"])
+
+    # Advanced automatic
+    result = await hass.config_entries.flow.async_init(
+        config_flow.DOMAIN, context={"source": "user", "show_advanced_options": True}
+    )
+
+    assert result["data_schema"] is not None
+    assert result["type"] == "form"
+    assert result["step_id"] == "user_advanced"
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"], user_input={"setup_method": AUTOMATIC_SETUP_STRING}
+    )
+
+    assert result["type"] == "external"
+    hass.config_entries.flow.async_abort(result["flow_id"])
+
+    # Advanced manual
+    result = await hass.config_entries.flow.async_init(
+        config_flow.DOMAIN, context={"source": "user", "show_advanced_options": True}
+    )
+
+    assert result["data_schema"] is not None
+    assert result["type"] == "form"
+    assert result["step_id"] == "user_advanced"
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"], user_input={"setup_method": MANUAL_SETUP_STRING}
+    )
+
+    assert result["type"] == "form"
+    assert result["step_id"] == "manual_setup"
+
+    mock_plex_server = MockPlexServer()
+
+    MANUAL_SERVER = {
+        CONF_HOST: MOCK_SERVERS[0][CONF_HOST],
+        CONF_PORT: MOCK_SERVERS[0][CONF_PORT],
+        CONF_SSL: False,
+        CONF_VERIFY_SSL: True,
+        CONF_TOKEN: MOCK_TOKEN,
+    }
+
+    MANUAL_SERVER_NO_HOST_OR_TOKEN = {
+        CONF_PORT: MOCK_SERVERS[0][CONF_PORT],
+        CONF_SSL: False,
+        CONF_VERIFY_SSL: True,
+    }
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"], user_input=MANUAL_SERVER_NO_HOST_OR_TOKEN
+    )
+
+    assert result["type"] == "form"
+    assert result["step_id"] == "manual_setup"
+    assert result["errors"]["base"] == "host_or_token"
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5462445)